presto-main/src/test/java/io/prestosql/sql/planner/TestUnwrapCastInComparison.java

```diff
@@ -977,4 +1053,29 @@ public void testNoEffect()
                         filter("CAST(A AS INTEGER) = 1",
                                 values("A"))));
     }
+
+    private void testNoUnwrap(Session session, String inputType, String inputPredicate, String expectedCastType)
```

Comment:
I started with explicit tests, following other code in this class. Change the approach because:

- having 5 lines for each test case felt overly verbose in a test method when i plan to cover many individual test cases (eg timestamps case)
   - with that many cases it's important to be able to compare between them to see what's tested, what's not, so verbosity matters here
- after extracting the method the test code provides what is important -- the symbol type, the constant compared against and the expected state -- without providing irrelevant pieces like VALUES 
   - one day we will add constant folding for Filter on top of VALUES ...
   - or we will just add additional  assertion where the VALUES are replaced with a temporary test table in eg memory connector

>  but the test cases are self-evident, easier to understand at first glance, and easier to copy/paste when testing with a live server.

i agree with this